modules/compute/src/test/java/org/apache/ignite/internal/compute/executor/ComputeExecutorTest.java

```diff
@@ -114,4 +119,61 @@ public Integer execute(JobExecutionContext context, Object... args) {
         }
     }
 
+    @Test
+    void retryJobFail() {
+        AtomicInteger runTimes = new AtomicInteger();
+
+        int maxRetries = 5;
+
+        JobExecution<Integer> execution = computeExecutor.executeJob(
+                new ExecutionOptions(0, maxRetries),
+                RetryJobFail.class,
+                new Object[]{runTimes}
+        );
+
+        await().until(execution::status, jobStatusWithState(FAILED));
+
+        assertThat(runTimes.get(), is(maxRetries + 1));
+    }
+
+    private static class RetryJobFail implements ComputeJob<Integer> {
+
+        @Override
+        public Integer execute(JobExecutionContext context, Object... args) {
+            AtomicInteger runTimes = (AtomicInteger) args[0];
+            runTimes.incrementAndGet();
+            throw new RuntimeException();
+        }
+    }
+
+    @Test
+    void retryJobSuccess() {
+        AtomicInteger runTimes = new AtomicInteger();
+
+        int maxRetries = 5;
+
+        JobExecution<Integer> execution = computeExecutor.executeJob(
+                new ExecutionOptions(0, maxRetries),
+                RetryJobSuccess.class,
+                new Object[]{runTimes, maxRetries}
+        );
+
+        await().until(execution::status, jobStatusWithState(COMPLETED));
+
+        assertThat(runTimes.get(), is(maxRetries + 1));
+    }
```

Comment:
Now it can be overtesting but in the future, this test can prevent tricky bugs. I am talking about checking the happy path when the job just passes and it is executed exactly once.